stakingModel/contracts/InterestDistribution.sol

```diff
@@ -29,6 +29,7 @@ library InterestDistribution {
         uint256 lastStake;
         uint256 withdrawnToDate;
         uint256 stakeBuyinRate;  // Precision points = 27 + 2 (G$ decimal) = 29
+        uint256 interestClaimedBeforeLastUnstake;
```

Comment:
i dont understand why we need this

Reply:
While unstake, we are resetting withdrawnToDate so to preserve staker's all time claimed interest we need one variable. Let me know if we don't need to preserve staker's all time claimed interest, will  remove it.

Reply:
we dont need it.
you can merge PRs after you remove it

Reply:
removed